src/vm/threads.cpp

```diff
@@ -8332,32 +8358,44 @@ LONG Thread::GetTotalThreadPoolCompletionCount()
     }
     CONTRACTL_END;
 
-    LONG total;
-    if (g_fEEStarted) //make sure we actually have a thread store
-    {
-        // make sure up-to-date thread-local counts are visible to us
-        ::FlushProcessWriteBuffers();
+    // enumerate all threads, summing their local counts.
+    ThreadStoreLockHolder tsl;
 
-        // enumerate all threads, summing their local counts.
-        ThreadStoreLockHolder tsl;
+    UINT64 total = GetOverflowCount(overflowCount);
```

Comment:
I guess getting the overflow count first and then iterating through the threads ensures we won't double-count a thread's stats if the thread is currently being removed?

Reply:
Getting the total count and updating the overflow count when a thread is removed, are both done inside the thread store lock, so the order shouldn't matter.